tools/accuracy_checker/accuracy_checker/data_readers/data_reader.py

```diff
@@ -180,8 +179,163 @@ def read(self, data_id):
+                image.putpalette(np.asarray(pal, dtype=np.uint8).tostring())
+                # Becomes a mode='P' automagically.
+            elif mode == 'P':  # default gray-scale
+                pal1 = np.arange(0, 256, 1, dtype=np.uint8)[:, np.newaxis]
+                pal2 = np.ones((3,), dtype=np.uint8)[np.newaxis, :]
+                pal = pal1 * pal2
+                image.putpalette(np.asarray(pal, dtype=np.uint8).tostring())
+
+            return image
+        if mode == '1':  # high input gives threshold for 1
+            bytedata = (data > high)
+            image = Image.frombytes('1', shape, bytedata.tostring())
+
+            return image
+
+        cmin = cmin or np.amin(np.ravel(data))
+        cmax = cmax or np.amax(np.ravel(data))
+        data = (data * 1.0 - cmin) * (high - low) / (cmax - cmin) + low
+        if mode == 'I':
+            data32 = data.astype(np.uint32)
+            image = Image.frombytes(mode, shape, data32.tostring())
+        else:
+            raise ValueError("Mode is unknown or incompatible with input array shape.")
+
+        return image
+
+    @staticmethod
+    def _process_3d(data, shape, mode, channel_axis, high, low, cmin, cmax):
+        # if here then 3-d array with a 3 or a 4 in the shape length.
+        # Check for 3 in datacube shape --- 'RGB' or 'YCbCr'
+        if channel_axis is None:
+            ca = np.flatnonzero(np.asarray(shape) == 3) if 3 in shape else np.flatnonzero(np.asarray(shape) == 4)
+            if not np.size(ca):
+                raise ValueError("Could not find channel dimension.")
+            ca = ca[0]
+        else:
+            ca = channel_axis
+
+        numch = shape[ca]
+        if numch not in [3, 4]:
+            raise ValueError("Channel axis dimension is not valid.")
+
+        bytedata = ScipyImageReader._bytescale(data, high=high, low=low, cmin=cmin, cmax=cmax)
+        channel_axis_mapping = {
+            0: ((1, 2, 0), (shape[1], shape[0])),
+            1: ((0, 2, 1), (shape[2], shape[0])),
+            2: ((0, 1, 2), (shape[1], shape[0]))
+        }
+        if ca in channel_axis_mapping:
+            transposition, shape = channel_axis_mapping[ca]
+            strdata = np.transpose(bytedata, transposition).tostring()
+
+        if mode is None:
+            mode = 'RGB' if numch == 3 else 'RGBA'
+
+        if mode not in ['RGB', 'RGBA', 'YCbCr', 'CMYK']:
+            raise ValueError("Mode is unknown or incompatible with input array shape.")
+
+        if mode in ['RGB', 'YCbCr'] and numch != 3:
+            raise ValueError("Invalid array shape for mode.")
+        if mode in ['RGBA', 'CMYK'] and numch != 4:
+            raise ValueError("Invalid array shape for mode.")
+
+        # Here we know data and mode is correct
+        image = Image.frombytes(mode, shape, strdata)
+        return image
+
+    @staticmethod
+    def _to_image(arr, high=255, low=0, cmin=None, cmax=None, pal=None, mode=None, channel_axis=None):
```

Comment:
None of these keyword parameters ever get set to anything but default values, so they can all be removed.